sql/core/src/main/scala/org/apache/spark/sql/jdbc/PostgresDialect.scala

```diff
@@ -382,18 +382,21 @@ private case class PostgresDialect()
       case Types.ARRAY =>
         val tableName = rsmd.getTableName(columnIdx)
         val columnName = rsmd.getColumnName(columnIdx)
-        val query =
+        val query = if (conf.postgresProperReadingOfArrayDimensionality) {
+             s"SELECT array_ndims($columnName)) FROM $tableName LIMIT 1"
+        } else {
           s"""
              |SELECT pg_attribute.attndims
              |FROM pg_attribute
```

Comment:
do you mean there is a bug in pgsql information schema which returns wrong `attndims`?

Reply:
interesting, let's leave a code comment here to explain it.

Reply:
Done